src/lib.rs

```diff
@@ -779,31 +779,37 @@ fn perform_reductions(
     try_alpha_reductions(png, &opts.alphas, eval);
 }
 
-/// Keep track of processing timeout
-pub(crate) struct Deadline {
+struct DeadlineImp {
```

Comment:
Does Imp = Implementation? I think `Impl` would be a more easily understood abbreviation. Alternatively it's common in Rust to name it something like `DeadlineInner`.

Reply:
I found `imp` pretty common in Rust land, because you can't use `impl` as a field name, and it's better to have consistency between type and field here. `Inner` works too I guess.